source/extensions/common/wasm/wasm.h

```diff
@@ -479,15 +517,21 @@ inline Context::Context(Wasm* wasm) : wasm_(wasm), id_(wasm->allocContextId()) {
 
 // Forward declarations for VM implemenations.
 template <typename R, typename... Args>
-void registerCallbackWavm(WasmVm* vm, absl::string_view functionName, R (*)(Args...));
+void registerCallbackWavm(WasmVm* vm, absl::string_view moduleName, absl::string_view functionName,
+                          R (*)(Args...));
 template <typename R, typename... Args>
 void getFunctionWavm(WasmVm* vm, absl::string_view functionName,
                      std::function<R(Context*, Args...)>*);
 
+template <typename T>
+std::unique_ptr<Global<T>> makeGlobalWavm(WasmVm* vm, absl::string_view moduleName,
```

Comment:
A naive question
Where is makeGlobalWavm instantiated?

Reply:
It is in the wasm/wavm/wavm.cc file. It is encapsulated in that file.  The idea is to make the details of the individual VM implementations private. I choose that because wavm/wavm.h is included in wasm/wasm.cc but not wasm/wasm.h to prevent circularizing the includes.  I am planning on moving class WavmVm into a new file and reorganizing the includes here which would enable moving this declaration into the wavm/ directory, after clearing the current PR backlog.